kernel/patches/4.19.x/0006-net-virtio_vsock-Fix-race-condition-between-bind-and.patch

```diff
@@ -0,0 +1,49 @@
+From 72e44f5a36d7226f105c9e71a52aa5322a561609 Mon Sep 17 00:00:00 2001
+From: Sebastien Boeuf <sebastien.boeuf@intel.com>
+Date: Thu, 13 Feb 2020 08:50:38 +0100
+Subject: [PATCH] net: virtio_vsock: Fix race condition between bind and listen
+
+Whenever the vsock backend on the host sends a packet through the RX
+queue, it expects an answer on the TX queue. Unfortunately, there is one
+case where the host side will hang waiting for the answer and will
+effectively never recover.
+
+This issue happens when the guest side starts binding to the socket,
+which insert a new bound socket into the list of already bound sockets.
+At this time, we expect the guest to also start listening, which will
+trigger the sk_state to move from TCP_CLOSE to TCP_LISTEN. The problem
+occurs if the host side queued a RX packet and triggered an interrupt
+right between the end of the binding process and the beginning of the
+listening process. In this specific case, the function processing the
+packet virtio_transport_recv_pkt() will find a bound socket, which means
+it will hit the switch statement checking for the sk_state, but the
+state won't be changed into TCP_LISTEN yet, which leads the code to pick
+the default statement. This default statement will only free the buffer,
+while it should also respond to the host side, by sending a packet on
+its TX queue.
+
+In order to simply fix this unfortunate chain of events, it is important
+that in case the default statement is entered, and because at this stage
+we know the host side is waiting for an answer, we must send back a
+packet containing the operation VIRTIO_VSOCK_OP_RST.
+
+Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
+---
+ net/vmw_vsock/virtio_transport_common.c | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/net/vmw_vsock/virtio_transport_common.c b/net/vmw_vsock/virtio_transport_common.c
+index 2a8651aa90c8..ccc40dd1771b 100644
+--- a/net/vmw_vsock/virtio_transport_common.c
++++ b/net/vmw_vsock/virtio_transport_common.c
+@@ -1051,6 +1051,7 @@ void virtio_transport_recv_pkt(struct virtio_vsock_pkt *pkt)
```

Comment:
The build for the kernel 4.19 is not working 
```
+ make -s CONFIG_DEBUG_SECTION_MISMATCH=y -j4 ARCH=x86_64
[ 1416s] net/vmw_vsock/virtio_transport_common.c: In function 'virtio_transport_recv_pkt':
[ 1416s] net/vmw_vsock/virtio_transport_common.c:1054:40: error: 't' undeclared (first use in this function); did you mean 'tm'?
[ 1416s]  1054 |   (void)virtio_transport_reset_no_sock(t, pkt);
[ 1416s]       |                                        ^
[ 1416s]       |                                        tm
[ 1416s] net/vmw_vsock/virtio_transport_common.c:1054:40: note: each undeclared identifier is reported only once for each function it appears in
[ 1416s] net/vmw_vsock/virtio_transport_common.c:1054:9: error: too many arguments to function 'virtio_transport_reset_no_sock'
[ 1416s]  1054 |   (void)virtio_transport_reset_no_sock(t, pkt);
[ 1416s]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 1416s] net/vmw_vsock/virtio_transport_common.c:663:12: note: declared here
[ 1416s]   663 | static int virtio_transport_reset_no_sock(struct virtio_vsock_pkt *pkt)
[ 1416s]       |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 1416s] make[2]: *** [scripts/Makefile.build:304: net/vmw_vsock/virtio_transport_common.o] Error 1
[ 1416s] make[1]: *** [scripts/Makefile.build:544: net/vmw_vsock] Error 2
[ 1416s] make: *** [Makefile:1052: net] Error 2
[ 1416s] make: *** Waiting for unfinished jobs....
```

https://build.opensuse.org/package/show/home:katacontainers:ci:x86_64:packaging-PR-933/linux-container